lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -2103,7 +2189,13 @@ CmpFOp::inferReturnTypes(MLIRContext *context, std::optional<Location> location,
                          mlir::RegionRange regions,
                          SmallVectorImpl<mlir::Type> &inferredReturnTypes) {
   OpBuilder builder(context);
-  inferredReturnTypes.push_back(ChannelType::get(builder.getIntegerType(1)));
+  inferredReturnTypes.push_back(ChannelType::get(
+      builder.getIntegerType(1),
+      // We can assume that
+      // - operand[0] exists
+      // - operand[0] is a channel type
+      // - all operands have the same extra signals
+      cast<ChannelType>(operands[0].getType()).getExtraSignals()));
```

Comment:
I added assertions for the first two, but the last one is difficult to assert.
I left a comment for the last one.

Reply:
yes I would prefer the infer functions to do as little verification as possible, but casting is so dangerous that we basically have no choice- if this cast can return a null pointer we should make a variable declaration and assert non-null, but if its an MLIR cast than we are fine

Reply:
The first two assertions seem less crucial? If they return a weird IR I feel the predicate's should check it, the assertions should only be to prevent segmentation fault, imo.